server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -366,17 +369,19 @@ protected void parseCreateField(ParseContext context, List<IndexableField> field
 
         // convert to utf8 only once before feeding postings/dv/stored fields
         final BytesRef binaryValue = new BytesRef(value);
-        if (fieldType().indexOptions() != IndexOptions.NONE || fieldType().stored()) {
+        if (fieldType().indexOptions() != IndexOptions.NONE || fieldType().stored())  {
             Field field = new Field(fieldType().name(), binaryValue, fieldType());
             fields.add(field);
+
+            if (!fieldType().hasDocValues() && fieldType().omitNorms()) {
```

Comment:
nit: we prefer to use `fieldType().hasDocValues() == false` for clarity.

Reply:
Got it. I will struggle a bit with this inside but codebase consistency is most important :)